internal/provider/resource_subaccount_trust_configuration.go

```diff
@@ -172,8 +172,12 @@ func (rs *subaccountTrustConfigurationResource) Read(ctx context.Context, req re
 		return
 	}
 
-	cliRes, _, err := rs.cli.Security.Trust.GetBySubaccount(ctx, state.SubaccountId.ValueString(), state.Origin.ValueString())
+	cliRes, rawRes, err := rs.cli.Security.Trust.GetBySubaccount(ctx, state.SubaccountId.ValueString(), state.Origin.ValueString())
 	if err != nil {
+		if rawRes.StatusCode == 404 {
```

Comment:
Can you move this special error handling in front of the generic error handling to avoid the nesting and make it easier to understand.

In addition, please add a comment to the Handling of the status code 404 as in case of the plugin dev best practices you referenced in the PR